app/src/main/java/org/mozilla/fenix/settings/account/AccountSettingsFragment.kt

```diff
@@ -218,7 +217,53 @@ class AccountSettingsFragment : PreferenceFragmentCompat() {
         )
     }
 
-    private fun showPinDialogWarning(newValue: Boolean) {
+    /**
+     * Prompts the user if they do not have a password/pin set up to secure their device, and
+     * updates the state of the sync engine with the new checkbox value.
+     *
+     * Currently used for logins and credit cards.
+     *
+     * @param newValue the value denoting whether or not to sync the specified preference.
+     */
+    private fun CheckBoxPreference.updateSyncEngineStateWithPinWarning(newValue: Boolean) {
+        val manager =
+            activity?.getSystemService(Context.KEYGUARD_SERVICE) as KeyguardManager
+        if (manager.isKeyguardSecure ||
+            !newValue ||
+            !requireContext().settings().shouldShowSecurityPinWarningSync
+        ) {
+            updateSyncEngineState(context, SyncEngine.CreditCards, newValue)
```

Comment:
🚨 We hardcoded `SyncEngine.CreditCards`
You need to pass the syncEngine as a variable otherwise, we always changing CreditCards.